x/rollapp/keeper/registered_denoms.go

```diff
@@ -32,3 +32,12 @@ func (k Keeper) IterateRegisteredDenoms(ctx sdk.Context, rollappID string, cb fu
 		return cb(item.K2())
 	})
 }
+
+func (k Keeper) ClearRegisteredDenoms(ctx sdk.Context, rollappID string) error {
+	return k.IterateRegisteredDenoms(ctx, rollappID, func(denom string) (bool, error) {
+		if err := k.registeredRollappDenoms.Remove(ctx, collections.Join(rollappID, denom)); err != nil {
+			return false, fmt.Errorf("clear registered denoms: %w", err)
+		}
+		return false, nil
+	})
+}
```

Comment:
you not reaaaaaaaaly supposed to delete while iterating in sdk afaiu

```suggestion
func (k Keeper) ClearRegisteredDenoms(ctx sdk.Context, rollappID string) error {
	rng := collections.NewPrefixedPairRange[string, string](rollappID)
	return k.registeredRollappDenoms.Clear(ctx, rng)
}

```